src/materialized/tests/sql.rs

```diff
@@ -878,11 +878,15 @@ fn test_tail_table_rw_timestamps() -> Result<(), Box<dyn Error>> {
 
     // Keep trying until you either panic or are able to verify the expected behavior.
     loop {
+        client_interactive.execute("BEGIN", &[])?;
```

Comment:
Why do these need to be added?

Reply:
The reason is because of this line: 
```
        // Two writes with the same data have their diffs compacted
        assert_eq!(rows[0].get::<_, Option<i64>>("mz_diff"), Some(2));
```
I believe that this is testing that two writes at the same timestamp will be compacted into a single write. However since every write now increases the timestamp, they were no longer being compacted. Including both writes in a transaction ensured that they'd happen on the same timestamp and be compacted.

I could of course be misunderstanding this test.

Reply:
Ah right, yes.